frontend/scripts/react-components/data/data-portal-page.component.jsx

```diff
@@ -0,0 +1,531 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React, { Component } from 'react';
+import _ from 'lodash';
+import { GET_CSV_DATA_DOWNLOAD_FILE, GET_JSON_DATA_DOWNLOAD_FILE, getURLFromParams } from 'utils/getURLFromParams';
+import BulkDownloadsBlock from 'react-components/data/bulk-downloads-block.component';
+import DownloadSelector from 'react-components/data/download-selector.component';
+import PropTypes from 'prop-types';
+import DataPortalDisabledMessage from 'react-components/data/data-portal-disabled-message.component';
+import DataPortalForm from 'react-components/data/data-portal-form.component';
+import classnames from 'classnames';
+
+class DataContent extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedContextId: null,
+      formVisible: false,
+
+      selectedCountry: null,
+      selectedCommodity: null,
+      selectedYears: [],
+      selectedExporters: [],
+      selectedConsumptionCountries: [],
+      selectedIndicators: [],
+
+      allYearsSelected: false,
+      allExportersSelected: false,
+      allConsumptionCountriesSelected: false,
+      allIndicatorsSelected: false,
+
+      outputType: 'pivot',
+      fileExtension: '.csv',
+      fileSeparator: 'comma',
+
+      currentDownloadParams: null,
+      currentDownloadType: null,
+
+      downloaded: false
+    };
```

Comment:
This component feels super bloated, the methods are doing too much for a component. I'd consider extracting the logic to the container and the store where possible. I know that there's a lot of different (and edge) cases we need to cover but it's pretty hard to follow the code.